tests/cmaes_test.cpp

```diff
@@ -17,6 +18,76 @@
 using namespace ens;
 using namespace ens::test;
 
+template<typename MatType = arma::mat, typename OptimizerType>
+void LogisticRegressionFunctionTestReal(OptimizerType& optimizer,
+                                        const double trainAccuracyTolerance,
+                                        const double testAccuracyTolerance,
+                                        const std::string dataset_path,
+                                        const size_t trials = 1,
+                                        const double split_ratio = 0.8)
+{
+  MatType data, testData, shuffledData;
+  arma::Row<size_t> responses, testResponses, shuffledResponses;
+
+  // Load the dataset
+  if (data.load(dataset_path, arma::csv_ascii) == false)
+  {
+    FAIL("couldn't load data");
+    return;
+  }
+  
+  data = data.t();
+  responses = arma::conv_to<arma::Row<size_t>>::from(data.row(0));
+  data = data.rows(1, data.n_rows-1);
+
+  arma::uvec indices = arma::shuffle(arma::linspace<arma::uvec>(0, 
+      data.n_cols - 1, data.n_cols)); 
+  size_t idx = data.n_cols * split_ratio;
+
+  shuffledData = MatType(data.n_rows, idx);
+  shuffledResponses = arma::Row<size_t>(idx);
+
+  testData = MatType(data.n_rows, data.n_cols-idx);
+  testResponses = arma::Row<size_t>(data.n_cols-idx);
+
+  for (size_t i = 0; i < idx; ++i)
+  {
+    shuffledData.col(i) = data.col(indices(i));
+    shuffledResponses.col(i) = responses.col(indices(i));
+  }
+  
+  for (size_t i = idx; i < data.n_cols; ++i)
+  {
+    testData.col(i-idx) = data.col(indices(i));
+    testResponses.col(i-idx) = responses.col(indices(i));
+  }
+
+  ens::test::LogisticRegression<MatType> lr(data, responses, 0.5);
+
+  for (size_t i = 0; i < trials; ++i)
+  {
+    MatType coordinates = lr.GetInitialPoint();
+    optimizer.Optimize(lr, coordinates);
+
+    const double acc = lr.ComputeAccuracy(shuffledData, shuffledResponses, 
+        coordinates);
+    const double testAcc = lr.ComputeAccuracy(testData, testResponses,
+        coordinates);
+
+    // Provide a shortcut to try again if we're not on the last trial.
+    if (i != (trials - 1))
+    {
+      if (acc != Approx(60.393258427).epsilon(trainAccuracyTolerance))
+        continue;
+      if (testAcc != Approx(66.4804).epsilon(testAccuracyTolerance))
+        continue;
+    }
+    REQUIRE(acc == Approx(60.393258427).epsilon(trainAccuracyTolerance));
+    REQUIRE(testAcc == Approx(66.4804).epsilon(testAccuracyTolerance));
```

Comment:
No need to compare this against a specifc accuracy, we should make sure it's above a certain level, so we could rewrite this as:

`REQUIRE(acc > 60);`

The same applies to the other statements.

Reply:
I tried the titanic test case for implemented CMAES in ensmallen and the results are not consistent and poor, this may due to the fact that the titanic dataset does not have enough dimension and a lot of categorial variables. So I tried on another dataset: Gisette (Nearly 5000 numerical dimensions and logistic regression on classifying digits). This one then all variants perform around the same rate. So I decided to change from titanic logistic regression to gisette regression. The problem is that this dataset is quite large in size compared to other datasets in the tests folder. So I am reducing the dataset to only around 1000 dimension - but it does not hurt the performance (I tested on range 200-5000 dimensions, all variants perform well on above 100 dimensions).

More information about Gisette dataset is here: https://archive.ics.uci.edu/ml/datasets/Gisette

Reply:
Now all test cases should be passed.